agent/src/main/java/com/teamscale/jacoco/agent/store/upload/UploadStoreBase.java

```diff
@@ -53,33 +48,36 @@ public UploadStoreBase(TimestampedFileStore failureStore, HttpUrl uploadUrl, Lis
 	protected abstract T getApi(Retrofit retrofit);
 
 	/** Uploads the coverage zip to the server */
-	protected abstract Response<ResponseBody> uploadCoverageZip(byte[] zipFileBytes) throws IOException, UploadStoreException;
+	protected abstract Response<ResponseBody> uploadCoverageZip(
+			File coverageFile) throws IOException, UploaderException;
 
 	@Override
-	public void store(String xml) {
+	public void upload(File coverageFile) {
 		try (Benchmark ignored = new Benchmark("Uploading report via HTTP")) {
-			if (!tryUpload(xml)) {
-				logger.warn("Storing failed upload in {}", failureStore.getOutputDirectory());
-				failureStore.store(xml);
+			if (!tryUpload(coverageFile)) {
+				logger.warn("Failed to upload coverage. Won't delete local file");
+			} else {
+				coverageFile.delete();
 			}
 		}
 	}
 
 	/** Performs the upload and returns <code>true</code> if successful. */
-	protected boolean tryUpload(String xml) {
+	protected boolean tryUpload(File coverageFile) {
 		logger.debug("Uploading coverage to {}", uploadUrl);
 
-		byte[] zipFileBytes;
+		File zipFile;
 		try {
-			zipFileBytes = createZipFile(xml);
+			zipFile = createZipFile(coverageFile);
 		} catch (IOException e) {
 			logger.error("Failed to compile coverage zip file for upload to {}", uploadUrl, e);
 			return false;
 		}
 
 		try {
-			Response<ResponseBody> response = uploadCoverageZip(zipFileBytes);
+			Response<ResponseBody> response = uploadCoverageZip(zipFile);
 			if (response.isSuccessful()) {
+				zipFile.delete();
 				return true;
 			}
 
```

Comment:
comment for the error we log later on (GitHub won't let me add comments in non-changed lines :-( ):
in case no error body is available, we currently log a newline and then nothing. This looks ugly in the logs. I'd prefer to log `<no server response>` or something like that in case the server does not return any body text to make it clear to the user what is happening